app/models/reservation.rb

```diff
@@ -0,0 +1,78 @@
+class Reservation < ApplicationRecord
+
+  belongs_to :branch
+  belongs_to :customer
+
+  has_and_belongs_to_many :tables
+  has_many :line_items, dependent: :destroy, after_add: :increase_total, after_remove: :decrease_total
```

Comment:
what happens if i update the line item's quantity. how will you handle that?

Reply:
Handled in another way.